csharp/class.go

```diff
@@ -77,23 +78,33 @@ func (self *ClassDeclaration) Property(type_ string, name string) *PropertyDecla
 	return property
 }
 
+func (self *ClassDeclaration) Summary(summary string) *ClassDeclaration {
+	self.summary = Summary(summary)
+	return self
+}
+
 func Class(name string) *ClassDeclaration {
 	return &ClassDeclaration{
 		name:       name,
 		modifiers:  []string{},
 		attributes: []Writable{},
 		members:    []Writable{},
+		summary:    nil,
 	}
 }
 
 func (self *ClassDeclaration) WriteCode(writer CodeWriter) {
+	if self.summary != nil {
```

Comment:
for example you do not need to do this kind of check. The logic for what the summary should should be pushed down into that struct. if the values are empty no need todo anything.